src/server/requestHandler.ts

```diff
@@ -124,24 +124,16 @@ export function createRequestHandler(options: CreateRequestHandlerOptions) {
 
     // API
     if (requestUrl.pathname.startsWith("/api")) {
-      const apiPaths = new Map([...raw, ...transpile]);
       const importAPIRoute = async (pathname: string): Promise<APIHandler> => {
-        let path = `${sourceDirectory}${pathname}`;
-        if (apiPaths.has(`${path}.js`)) {
-          path = `file://${cwd}/${path}.js`;
-        } else if (apiPaths.has(`${path}.ts`)) {
-          path = `file://${cwd}/${path}.ts`;
-        } else if (apiPaths.has(`${path}/index.js`)) {
-          path = `file://${cwd}/${path}/index.js`;
-        } else if (apiPaths.has(`${path}/index.ts`)) {
-          path = `file://${cwd}/${path}/index.ts`;
-        }
```

Comment:
All this is now handled by `apiRoutes.getHandler` with added support for dynamic routes (ie. routes with params).